src/main/java/duke/main/Storage.java

```diff
@@ -0,0 +1,152 @@
+package duke.main;
+
+import duke.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+
+/**
+ * Handle storing and retrieving of tasks from a file.
+ */
+public class Storage {
+    private final File saveFile;
+
+    /**
+     * Constructor for creating new save file.
+     * @param filePath Path where the saved file is located.
+     * @throws DukeException When error occur creating file
+     * @throws IOException When error occur creating file
+     */
+    public Storage(String filePath) throws DukeException, IOException {
+        String [] path = filePath.split("/");
+        File directory = new File(path[0]);
+        // Check if directory exists
+        if (!directory.exists()) {
+            // Does not exist, create new directory
+            if (!directory.mkdir()) {
+                throw new DukeException("Error encountered when creating folder to store save file");
+            }
+        }
+
+        File savedTasks = new File(filePath);
+        // Check if File exists
+        if (!savedTasks.exists()) {
+            // Does not exist, create new file
+            if (!savedTasks.createNewFile()) {
+                throw new DukeException("Error encountered when creating new save file");
+            }
+        }
+
+        this.saveFile = savedTasks;
+    }
```

Comment:
I like how the checking of the presence of an existing directory and file is implemented.